docs/pages/07_service.adoc

```diff
@@ -0,0 +1,1092 @@
+
+Both the return value and the parameter are required when calling the function if specified in the function declaration.
+
+The `throws` keyword is used to declare a list of errors that may occur during the execution of the function code. This optional error list contains the names of the errors that the function may throw and the caller must be prepared to handle them.
+
+The body of the function is optional. If no function body is specified (e.g. missing block with curly braces), the function must be implemented in Java.
+
+*Example:*
+
+----
+service PersonService(Person person) {
+    function void delete() ;
+    function LeadTransfer[] addLead(LeadTransfer lead) ;
+}
+----
+
+The above example shows two functions. The `delete` function has no return value or parameters and will be implemented in Java. The second function (`addLead`) has both a return value and a parameter. This must also be implemented in Java.
+
+=== Data function
+
+Data functions are used to query the data model quickly and elegantly. The data function is an expression (see Expression) that will be evaluated when the function is invoked. Data functions are very similar to queries (see Query), except that they have no input parameters and are used by external systems or users.
+
+A very important feature of data functions is that their results can be filtered and paging is provided out of the box. Filtering can be performed on the values of the return value fields. The aim of the paging is, dividing the result into discrete pages with the help of the function.
+
+The data function has a name and a return value. The return value is a computed value passed to the function caller.
+
+Use the `function` keyword to specify a data function.
+
+*Syntax:*
+
+----
+function <primitive> <name> => <expression> ;
+----
+
+where <primitive> is the name of a domain model primitive that will be returned, and <name> is the referable name of the function, or
+
+*Syntax:*
+
+----
+function <transfer>[[]] <name> => <expression> [options <options>];
+----
+
+where the <transfer> is the name of a mapped transfer object that will be returned, and <name> is the referable name of the function. The optional [] indicates that the function returns a collection of <transfer> instances rather than a single transfer object.
+
+The <expression> calculates the return value of the service whenever it is requested. See Expression. If the export that groups the service is a mapped export, the mapping field is available within the expression.
+
+The <options> is an optional expression. If it is specified, the result of the <options> expression must be a list of <transfer> objects. This is the list objects from which the user can select objects and insert then into the list specified by <expression>. If <options> is not specified, the list defined by <expression> cannot be changed.
```

Comment:
"select objects and insert then into the list specified by <expression>" Something does not add up. The paragraph started with "Data functions are used to query the data model quickly and elegantly". Is it one functionality or two baked in?